code/datums/components/squashable.dm

```diff
@@ -27,7 +29,7 @@
 	if(!src.on_squash_callback && squash_callback)
 		on_squash_callback = CALLBACK(parent, squash_callback)
 
-	AddElement(/datum/element/connect_loc_behalf, parent, loc_connections)
+	connect_loc_behalf = AddComponent(/datum/component/connect_loc_behalf, parent, loc_connections)
```

Comment:
This has to happen because components that can have duplicates cannot be retrieved with GetComponent or GetExactComponent.
I would have to use GetComponents, which returns a list of components which may look a bit more ugly because I'd be doing
```
var/list/list_of_components = GetComponents(type)
var/datum/component/component = list_of_components[1]
component.RemoveComponent()

Reply:
Additionally, if you want to disallow duplicates, that could be a possibility too.

Reply:
Having a signal to trigger deletion on the /datum/component/connect_loc_behalf seems really hacky.

Reply:
Additionally, you could have AddComponent pass arguments to delete the component but honestly that seems like something that would be very misleading